styx-scheduler-service/src/main/java/com/spotify/styx/flyte/FlyteAdminClientRunner.java

```diff
@@ -143,4 +147,28 @@ private int flyteErrorCodeToStyx(final String errorCode) {
           return UNKNOWN_ERROR_EXIT_CODE;
     }
   }
+
+  private ExecutionMode toFlyteExecutionMode(Trigger trigger) {
+    return trigger.accept(new TriggerVisitor<ExecutionMode>() {
+      @Override
+      public ExecutionMode natural() {
+        return ExecutionMode.SCHEDULED;
+      }
+
+      @Override
+      public ExecutionMode adhoc(String triggerId) {
+        return ExecutionMode.MANUAL;
+      }
+
+      @Override
+      public ExecutionMode backfill(String triggerId) {
+        return ExecutionMode.RELAUNCH; //TODO: verify that this means the same
```

Comment:
Should a retry also be defined as a RELAUNCH?

Reply:
I am not sure if should we include retries as `RELAUNCH`. I am thinking about we trigger the execution and it fails because of missing dependencies 5 times in a row and then succeeds. Even if we reach until retry number 5 I think that we still should count them all as `SCHEDULED` (or `MANUAL` if it was ad_hoc)

Reply:
After dicussing this with the team we decided that RELAUNCH implies a idempodem relaunch but styx doesn't offer such guaranties. So will never use RELAUNCH and backfills re turned into MANUAL